### ISSUE TITLE
(appsync): Add ability to add additional authorization mode on existing appsyn api

### ISSUE THREAD

**Issue by olehrdk** [aws/aws-cdk#30398]
### Describe the feature

Add support for additional authorization modes on existing app sync api (retrieved by calling `appsync.GraphqlApi.fromGraphqlApiAttributes`)

### Use Case

I have Amplify project with default Cognito user pool authorization mode. I want to add one more user pool as additional mode. Amplify does not support 2 user pools, that's why I use CDK to create one more user pool, and want to be able to add it as additional mode to my app sync api with CDK.

### Proposed Solution

Add new method (`addAdditionalAuthorizationModes`) on `IGraphqlApi` similar to `setupAdditionalAuthorizationModes` which will add additional authorizations modes to existing app sync api.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [X] This feature might incur a breaking change

### CDK version used

2.143.1

### Environment details (OS name and version, etc.)

-

**Comment by khushail** [comment]
Hi @olehrdk , thanks for reaching out. I am not very much clear about your ask here. I see these are the [supported authorization modes](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_appsync.AuthorizationMode.html) which includes cognito user pool and here is a [doc](https://docs.aws.amazon.com/appsync/latest/devguide/security-authz.html#using-additional-authorization-modes) on how you could use additional authorization modes. 
I highly doubt providing additional user pool would be a good idea here. @pahud ,wdyt?

**Comment by bschulz87** [comment]
I have the exact same requirement:

- AppSync created via Amplify for DataStore usage
- Multiple User Pools created via CDK

How to add these User Pools to the existing imported [GraphqlApi](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_appsync.GraphqlApi.html#methods).fromGraphqlApiAttributes?

**Comment by samson-keung** [comment]
Hi @olehrdk @bschulz87 , if I understood correctly, you have a `GraphqlApi` that is created via Amplify (not created in CDK). And you would like to import it into CDK and make changes to it.

Modifying imported resources (e.g. a `GraphqlApi` instance retrieved via the `fromGraphqlApiAttributes` method) is not supported in CDK. Such imported resource is a read-only object. I would recommend reaching out to Amplify support about how to add additional user pool.

**Comment by bschulz87** [comment]
Yes. I am aware that CDK is built upon CloudFormation which also has no way to add User Pools to an imported AppSync Resource. Once it is possible there i would still love to see the requested feature here.
